### PR TITLE
Script After Upgrade to node v14 1901

### LINKED ISSUE #1901: Support node v14 to compile Luigi
Currently we cannot compile Luigi in node v14

### PR DESCRIPTION
**Pull request by legteodav**
**Result:**
After changing node version, you would need to run the following command to update and link all the respective packages:
`npm run change-node-version`

**Explanation:**
Internally the script (node ./scripts/change.node.version.js) will:
1) Run `npm link @angular/cli`
2) Delete all node_modules dir
3) For all the modules, it will run `npm install`

 Resolves #1901

### CODE REVIEW

**Review by JohannesDoberer — APPROVED** [review]
Works like a charm 👍

File: scripts/change.node.version.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+const exec = require('child_process').exec;
+const fs = require('fs');
+const path = require('path');
+
+
+const listModules = () => {
+  const lernaPath =  path.join(path.resolve(__dirname), '..', 'lerna.json')
```
Comment (JohannesDoberer) [inline scripts/change.node.version.js:7]:
```suggestion
  const lernaPath =  path.join(path.resolve(__dirname), '..', 'lerna.json');
```

File: scripts/change.node.version.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+const exec = require('child_process').exec;
+const fs = require('fs');
+const path = require('path');
+
+
```
Comment (ndricimrr) [inline scripts/change.node.version.js:5]:
```suggestion
// return list of all module packages
```

File: scripts/change.node.version.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const listModules = () => {
+  const lernaPath =  path.join(path.resolve(__dirname), '..', 'lerna.json');
+  const lernaConfig = JSON.parse(fs.readFileSync(lernaPath).toString());
+  return lernaConfig.packages;
+}
+
```
Comment (ndricimrr) [inline scripts/change.node.version.js:11]:
```suggestion

// helper async function used to execute our 'npm' commands  
```

File: scripts/change.node.version.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        console.error('exec error: ' + error);
+        reject(stderr);
+        return;
+      }
+      console.log(stdout);
+      resolve(stdout);
+    });
+  });
+}
+
```
Comment (ndricimrr) [inline scripts/change.node.version.js:25]:
```suggestion

// takes a module as a parameter and deletes its dependencies and then re-installs them sequentially
```

File: scripts/change.node.version.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+const executeModule = async (module) => {
+    const pathArray = [path.resolve(__dirname),'..'].concat(module.split('/'));
+    const modulePath =  path.join(...pathArray);
+    const moduleDependencies =   path.join(modulePath, 'node_modules');
```
Comment (ndricimrr) [inline scripts/change.node.version.js:29]:
```suggestion
    const moduleDependencies =  path.join(modulePath, 'node_modules');
```
small space

File: scripts/change.node.version.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    console.log("Deleting directory "+moduleDependencies);
```
Comment (ndricimrr) [inline scripts/change.node.version.js:30]:
```suggestion
    console.log("Deleting directory " + moduleDependencies);
```
small spacing

File: scripts/change.node.version.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    fs.rmdirSync(moduleDependencies, { recursive: true });
+
+    process.chdir( modulePath )
+    console.log("npm install on directory "+modulePath);
```
Comment (ndricimrr) [inline scripts/change.node.version.js:34]:
```suggestion
    console.log("npm install on directory " + modulePath);
```
small space

File: scripts/change.node.version.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        return;
+      }
+      console.log(stdout);
+      resolve(stdout);
+    });
+  });
+}
+
+const executeModule = async (module) => {
+    const pathArray = [path.resolve(__dirname),'..'].concat(module.split('/'));
```
Comment (ndricimrr) [inline scripts/change.node.version.js:27]:
```suggestion
    // retrieve path from module
    const pathArray = [path.resolve(__dirname),'..'].concat(module.split('/'));
```

**Review by ndricimrr — APPROVED** [review]
Works pretty well for me too. 👍 

Just added some comments since these scripts tend to be hard to understand in the long run

**Pull request merged**